src/views/DatasetManager/components/CreateDatasetButton/components/DatasetCreationParameters/DatasetCreationParametersHook.js

```diff
@@ -0,0 +1,188 @@
+
+  // Each parameter must have a id strictly unique.
+  const hardCodedDataSources = useMemo(() => {
+    const dataSources = [
+      {
+        id: DATASET_SOURCE_TYPE.AZURE_STORAGE,
+        labels: {
+          en: 'Graph Format from Azure Storage',
+          fr: 'Format Graph depuis Azure Storage',
+        },
+        parameters: [
+          {
+            id: `name`,
+            varType: 'string',
+            labels: { en: 'Account name', fr: 'Nom du compte' },
+          },
+          {
+            id: `location`,
+            varType: 'string',
+            labels: { en: 'Container name', fr: 'Nom du container' },
+          },
+          {
+            id: `path`,
+            varType: 'string',
+            labels: { en: 'Path', fr: 'Chemin' },
+          },
+        ],
+      },
+      {
+        id: DATASET_SOURCE_TYPE.ADT,
+        labels: {
+          en: 'Azure Digital Twin',
+          fr: 'Azure Digital Twin',
+        },
+        parameters: [
+          {
+            id: `location`,
+            varType: 'string',
+            labels: { en: 'Path', fr: 'Chemin' },
+          },
+        ],
+      },
+      {
+        id: DATASET_SOURCE_TYPE.LOCAL_FILE,
+        labels: {
+          en: 'Graph Format from Local File',
+          fr: 'Format Graph depuis un fichier local',
+        },
+        parameters: [{ id: `file`, varType: '%DATASETID%', labels: { en: '', fr: '' } }],
+      },
+      {
+        id: DATASET_SOURCE_TYPE.NONE,
+        labels: {
+          en: 'Empty',
+          fr: 'Dataset vide',
+        },
+        parameters: [],
+      },
+    ];
+
+    dataSources.forEach((dataSource) => {
+      dataSource.parameters.forEach((parameter) => (parameter.id = `${dataSource.id}.${parameter.id}`));
+    });
+
+    TranslationUtils.addTranslationRunTemplateLabels(dataSources);
+    TranslationUtils.addTranslationParametersLabels(dataSources.flatMap((dataSource) => dataSource?.parameters));
+
+    return dataSources;
+  }, []);
+
+  const dataSourceRunTemplates = useMemo(() => {
+    const dataSources = solutionData.runTemplates.filter((runTemplate) => runTemplate?.tags.includes('datasource'));
+
+    const parameters = solutionData.parameters;
+    const runTemplatesParameters = solutionData.runTemplatesParametersIdsDict;
+
+    const dataSourcesWithParameters = dataSources.map((dataSource) => {
+      const dataSourceWithParameters = { ...dataSource };
+      dataSourceWithParameters.parameters = parameters.filter((parameter) =>
+        runTemplatesParameters[dataSource.id].includes(parameter.id)
+      );
+      return dataSourceWithParameters;
+    });
+
+    return [...hardCodedDataSources, ...dataSourcesWithParameters];
+  }, [
+    solutionData.parameters,
+    solutionData.runTemplates,
+    solutionData.runTemplatesParametersIdsDict,
+    hardCodedDataSources,
+  ]);
+
+  const dataSourceTypeEnumValues = useMemo(() => {
+    return [
+      ...dataSourceRunTemplates.map((dataSource) => {
```

Comment:
The `map` function already returns an array, different from the original one. Do we really need to have this new list and spread operator?